src/test/fuzz/cluster_linearize.cpp

```diff
@@ -0,0 +1,390 @@
+        VectorWriter writer(ser, 0);
+        writer << Using<DepGraphFormatter>(depgraph);
+        SpanReader reader(ser);
+        DepGraph<TestBitSet> decoded_depgraph;
+        reader >> Using<DepGraphFormatter>(decoded_depgraph);
+        assert(depgraph == decoded_depgraph);
+        assert(reader.empty());
+    }
+}
+
+} // namespace
+
+FUZZ_TARGET(clusterlin_add_dependency)
+{
+    // Verify that computing a DepGraph from a cluster, or building it step by step using AddDependency
+    // have the same effect.
+
+    // Construct a cluster of a certain length, with no dependencies.
+    Cluster<TestBitSet> cluster;
+    FuzzedDataProvider provider(buffer.data(), buffer.size());
+    auto num_tx = provider.ConsumeIntegralInRange<ClusterIndex>(2, 32);
+    cluster.resize(num_tx);
+    for (auto& item : cluster) item.first.size = 1;
+    // Construct the corresponding DepGraph object (also no dependencies).
+    DepGraph depgraph(cluster);
+    SanityCheck(depgraph);
+    // Read (parent, child) pairs, and add them to the cluster and txgraph.
+    LIMITED_WHILE(provider.remaining_bytes() > 0, 1024) {
+        auto parent = provider.ConsumeIntegralInRange<ClusterIndex>(0, num_tx - 1);
+        auto child = provider.ConsumeIntegralInRange<ClusterIndex>(0, num_tx - 2);
+        child += (child >= parent);
+        cluster[child].second.Set(parent);
+        depgraph.AddDependency(parent, child);
+        assert(depgraph.Ancestors(child)[parent]);
+        assert(depgraph.Descendants(parent)[child]);
+    }
+    // Sanity check the result.
+    SanityCheck(depgraph);
+    // Verify that the resulting DepGraph matches one recomputed from the cluster.
+    assert(DepGraph(cluster) == depgraph);
+}
+
+FUZZ_TARGET(clusterlin_cluster_serialization)
+{
+    // Verify that any graph of transaction has its ancestry correctly computed by DepGraph, and if
+    // it is a DAG, it can be serialized as a DepGraph in a way that roundtrips. This guarantees
+    // that any acyclic cluster has a corresponding DepGraph serialization.
+
+    FuzzedDataProvider provider(buffer.data(), buffer.size());
+
+    // Construct a cluster in a naive way (using a FuzzedDataProvider-based serialization).
+    Cluster<TestBitSet> cluster;
+    auto num_tx = provider.ConsumeIntegralInRange<ClusterIndex>(1, 32);
+    cluster.resize(num_tx);
+    for (ClusterIndex i = 0; i < num_tx; ++i) {
+        cluster[i].first.size = provider.ConsumeIntegralInRange<int32_t>(1, 0x3fffff);
+        cluster[i].first.fee = provider.ConsumeIntegralInRange<int64_t>(-0x8000000000000, 0x7ffffffffffff);
+        for (ClusterIndex j = 0; j < num_tx; ++j) {
+            if (i == j) continue;
+            if (provider.ConsumeBool()) cluster[i].second.Set(j);
+        }
+    }
+
+    // Construct dependency graph. The sanity check here includes a round-trip check.
+    DepGraph depgraph(cluster);
+    SanityCheck(depgraph);
+
+    // Verify that ancestry is computed correctly.
```

Comment:
for assurance shouldn't we also be checking the descendants are being computed as expected? Or is that covered somewhere else?